### PR TITLE
Feature editor indent option

### PR DESCRIPTION
**Pull request by allenyang79**
add indent setting on editor.
include tabSize,indentUnit,indentWithTabs
thanks

### CODE REVIEW

File: panel/repl.html [hunk 1]
```diff
@@ -40,10 +40,26 @@ <h3 class="type--standard-weight">Settings</h3>
...
         <div class="settings__content">
           <div class="settings-column">
-            <h4 .>Transformer</h4>
+            <h4>Transformer</h4>
             <div class="transformer-options"></div>
           </div>
           <div class="settings-column">
+            <h4>Editor</h4>
+            <div class="settings__option-container">
+                <div>tabSize</div>
```
Comment (richgilbank) [inline panel/repl.html]: These should probably be human-readable titles, i.e. `Tab size` and `Indent with tabs`

File: panel/scripts/repl.js [hunk 2]
```diff
@@ -43,7 +43,9 @@ Repl.prototype.onDomReady = function() {
     matchBrackets: true,
     continueComments: "Enter",
     extraKeys: {"Ctrl-Q": "toggleComment"},
-    tabSize: 2,
+    tabSize: this.settings.data.tabSize || 2,
+    indentUnit: this.settings.data.tabSize || 2,
+    indentWithTabs: true,//this.settings.data.indentWithTabs || false,
```
Comment (richgilbank) [inline panel/scripts/repl.js]: Looks like this got commented out in dev - should be uncommented.

File: panel/scripts/settings.js [hunk 3]
```diff
@@ -7,6 +7,9 @@ function Settings(repl) {
   this.repl = repl;
   this.DEFAULTS = this.data = {
     transformer: 'babel',
+    tabSize: 2,
+    indentUnit: 2,
+    indentWithTabs: true,
```
Comment (richgilbank) [inline panel/scripts/settings.js]: I'd like to keep this defaulted to false (use spaces), as that's how it currently behaves

**Comment by richgilbank** [comment]
Thanks for submitting this @allenyang79! I've commented on a few little things, but other than that, LGTM! Will merge when you can update with those changes.

**Comment by richgilbank** [comment]
Bump :)

**Comment by abhisekp** [comment]
@richgilbank could you please correct this PR? Seems @allenyang79 is unavailable.
Currently, the mixed tab and space is a PITA and this a much required feature.

cc: @alanbuchanan

**Comment by richgilbank** [comment]
Closing in favour of https://github.com/richgilbank/Scratch-JS/issues/76

**Pull request closed without merging**